cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java

```diff
@@ -114,30 +123,23 @@ public MetricsConsumerMetaTable createConsumerMeta() {
     }
   }
 
-  private MetricsTable getOrCreateMetricsTable(String tableName, DatasetProperties props) {
-    MetricsTable table = null;
+  private MetricsTable getOrCreateMetricsTable(String tableName, final DatasetProperties props) {
     // metrics tables are in the system namespace
-    DatasetId metricsDatasetInstanceId = NamespaceId.SYSTEM.dataset(tableName);
-    while (table == null) {
-      try {
-        table = DatasetsUtil.getOrCreateDataset(dsFramework, metricsDatasetInstanceId,
-                                                MetricsTable.class.getName(), props, null, null);
-      } catch (DatasetManagementException | ServiceUnavailableException e) {
-        // dataset service may be not up yet
-        // todo: seems like this logic applies everywhere, so should we move it to DatasetsUtil?
-        LOG.warn("Cannot access or create table {}, will retry in 1 sec.", tableName);
-        try {
-          TimeUnit.SECONDS.sleep(1);
-        } catch (InterruptedException ie) {
-          Thread.currentThread().interrupt();
-          break;
+    final DatasetId metricsDatasetInstanceId = NamespaceId.SYSTEM.dataset(tableName);
+    MetricsTable table;
+    try {
+      table = Retries.callWithRetries(new Retries.Callable<MetricsTable, Exception>() {
+        @Override
+        public MetricsTable call() throws Exception {
+          return DatasetsUtil.getOrCreateDataset(dsFramework, metricsDatasetInstanceId, MetricsTable.class.getName(),
+                                                 props, null, null);
         }
-      } catch (IOException e) {
-        LOG.error("Exception while creating table {}.", tableName, e);
-        throw Throwables.propagate(e);
-      }
+      }, RetryStrategies.fixDelay(1, TimeUnit.SECONDS), RETRYABLE_PREDICATE);
```

Comment:
@chtyim Addressed comments.
1. Not retrying in the `DefaultMetricDatasetFactory` instead relying on the caller.
2. In `OperationalStatsService`, catching the `ServiceUnavailableException` and not logging. This will be tried in the next iteration anyway.
3. We were already retrying by sleeping for 1 second in the `KafkaMetricsProcessorService` and `MessagingMetricsProcessorService`. Catching the `ServiceUnavailableException` there so that we do not log unnecessary.